### PR TITLE
Add OSPF summary-lsa metric

### LINKED ISSUE #915: Missing link metric in ospfv2 summary-lsa model
Hi all,

I have a question regarding the model for OSPFv2 Summary LSA.
Looking at the model for it:
```
+--ro summary-lsa
|  +--ro state
|  |  +--ro network-mask?   uint8 
|  +--ro types-of-service
|     +--ro type-of-service* [tos]
|        +--ro tos      -> ../state/tos
|        +--ro state
|           +--ro tos?      uint8 
|           +--ro metric?   oc-ospf-types:ospf-metric
```
There isn't any field explicitly for the link metric. The only metric field here is under TOS.

For reference, as-external-lsa made an explicit indication of link metric apart from the TOS-metric field:
```
+--ro as-external-lsa
|  +--ro state
|  |  +--ro mask?                 uint8
|  |  +--ro metric-type?          enumeration
|  |  +--ro metric?               oc-ospf-types:ospf-metric                 <<<
|  |  +--ro forwarding-address?   inet:ipv4-address-no-zone
|  |  +--ro external-route-tag?   uint32
|  +--ro types-of-service
|     +--ro type-of-service* [tos]
|        +--ro tos      -> ../state/tos
|        +--ro state
|           +--ro forwarding-address?   inet:ipv4-address-no-zone
|           +--ro external-route-tag?   uint32
|           +--ro tos?                  uint8
|           +--ro metric?               oc-ospf-types:ospf-metric
```

Is summary-lsa missing a field for the link metric? Or is the link metric expected to be under TOS-metric with the TOS value being 0?

Thanks in advance!

### PR DESCRIPTION
**Pull request by dplore**
### Change Scope

Add
`/network-instances/network-instance/protocols/protocol/ospfv2/areas/area/lsdb/lsa-types/lsa-type/lsas/lsa/summary-lsa/state/metric`

* OSPF summary LSA contains a metric field, but the OC model is missing this.
* This change is backwards compatible.

### Platform Implementations

 * [JunOS show ospf database](https://www.juniper.net/documentation/us/en/software/junos/ospf/topics/ref/command/show-ospf-database.html) 
 * [Cisco IOS XR show ospf database summary](https://www.cisco.com/c/en/us/td/docs/ios_xr_sw/iosxr_r3-7/routing/command/reference/rr37ospf.html#wp1226184)

# Tree view
```
        |     +--rw ospfv2
        |     |  +--rw areas
        |     |     +--rw area* [identifier]
        |     |        +--ro lsdb
        |     |        |  +--ro lsa-types
        |     |        |     +--ro lsa-type* [type]
        |     |        |        +--ro lsas
        |     |        |           +--ro lsa* [link-state-id]
        |     |        |              +--ro summary-lsa
        |     |        |              |  +--ro state
        |     |        |              |  |  +--ro network-mask?   uint8
        |     |        |              |  |  +--ro metric?         oc-ospf-types:ospf-metric      <-- added
        |     |        |              |  +--ro types-of-service
        |     |        |              |     +--ro type-of-service* [tos]
        |     |        |              |        +--ro tos      -> ../state/tos
        |     |        |              |        +--ro state
        |     |        |              |           +--ro tos?      uint8
        |     |        |              |           +--ro metric?   oc-ospf-types:ospf-metric
```

Fixes #915

### CODE REVIEW

File: release/models/bgp/openconfig-bgp-errors.yang [hunk 1]
```diff
@@ -18,7 +18,13 @@ submodule openconfig-bgp-errors {
     "This module defines BGP NOTIFICATION message error codes
     and subcodes";
 
-  oc-ext:openconfig-version "5.4.0";
+  oc-ext:openconfig-version "5.5.0";
+
+  revision "2023-06-05" {
+    description
+      "Add link bandwidth extended community type";
```
Comment (robshakir) [inline release/models/bgp/openconfig-bgp-errors.yang:25]: (nit) full stop on the end of this sentence.

File: release/models/bgp/openconfig-bgp-types.yang [hunk 2]
```diff
@@ -25,7 +25,13 @@ module openconfig-bgp-types {
     policy. It can be imported by modules that make use of BGP
     attributes";
 
-  oc-ext:openconfig-version "5.4.0";
+  oc-ext:openconfig-version "5.5.0";
+
+  revision "2023-06-08" {
+    description
+      "Add link bandwidth extended community type";
```
Comment (robshakir) [inline release/models/bgp/openconfig-bgp-types.yang:32]: As per above.

File: release/models/bgp/openconfig-bgp-types.yang [hunk 3]
```diff
@@ -579,6 +585,48 @@ module openconfig-bgp-types {
                 '429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[0-1][0-9]{8}|[1-3][0-9]{9}|'  +
                 '[1-9][0-9]{1,8}|[0-9])$';
       }
+      type string {
```
Comment (robshakir) [inline release/models/bgp/openconfig-bgp-types.yang:588]: This looks to be an unrelated change to the description. Incorrect files added in this branch perhaps?

File: release/models/ospf/openconfig-ospfv2-area-interface.yang [hunk 4]
```diff
@@ -25,7 +25,13 @@ submodule openconfig-ospfv2-area-interface {
     "This submodule provides OSPFv2 configuration and operational
     state parameters that are specific to the area context";
 
-  oc-ext:openconfig-version "0.4.2";
+  oc-ext:openconfig-version "0.5.0";
+
+  revision "2023-07-17" {
+    description
+      "add leaf metric to lsdb-summary-lsa-state";
```
Comment (robshakir) [inline release/models/ospf/openconfig-ospfv2-area-interface.yang:32]: Capitalise and make complete sentence.

File: release/models/ospf/openconfig-ospfv2-area.yang [hunk 5]
```diff
@@ -23,7 +23,13 @@ submodule openconfig-ospfv2-area {
     "This submodule provides OSPFv2 configuration and operational
     state parameters that are specific to the area context";
 
-  oc-ext:openconfig-version "0.4.2";
+  oc-ext:openconfig-version "0.5.0";
+
+  revision "2023-07-17" {
+    description
+      "add leaf metric to lsdb-summary-lsa-state";
```
Comment (robshakir) [inline release/models/ospf/openconfig-ospfv2-area.yang:30]: As above.

File: release/models/ospf/openconfig-ospfv2-common.yang [hunk 6]
```diff
@@ -17,7 +17,13 @@ submodule openconfig-ospfv2-common {
     "This submodule provides OSPFv2 configuration and operational
     state parameters that are shared across multiple contexts";
 
-  oc-ext:openconfig-version "0.4.2";
+  oc-ext:openconfig-version "0.5.0";
+
+  revision "2023-07-17" {
+    description
+      "add leaf metric to lsdb-summary-lsa-state";
```
Comment (robshakir) [inline release/models/ospf/openconfig-ospfv2-common.yang:24]: And again.

File: release/models/ospf/openconfig-ospfv2-global.yang [hunk 7]
```diff
@@ -23,7 +23,13 @@ submodule openconfig-ospfv2-global {
     "This submodule provides OSPFv2 configuration and operational
     state parameters that are global to a particular OSPF instance";
 
-  oc-ext:openconfig-version "0.4.2";
+  oc-ext:openconfig-version "0.5.0";
+
+  revision "2023-07-17" {
+    description
+      "add leaf metric to lsdb-summary-lsa-state";
```
Comment (robshakir) [inline release/models/ospf/openconfig-ospfv2-global.yang:30]: Ditto.

File: release/models/ospf/openconfig-ospfv2-lsdb.yang [hunk 8]
```diff
@@ -22,7 +22,13 @@ submodule openconfig-ospfv2-lsdb {
     "An OpenConfig model for the Open Shortest Path First (OSPF)
     version 2 link-state database (LSDB)";
 
-  oc-ext:openconfig-version "0.4.2";
+  oc-ext:openconfig-version "0.5.0";
+
+  revision "2023-07-17" {
+    description
+      "add leaf metric to lsdb-summary-lsa-state";
```
Comment (robshakir) [inline release/models/ospf/openconfig-ospfv2-lsdb.yang:29]: As above.

File: release/models/ospf/openconfig-ospfv2-lsdb.yang [hunk 9]
```diff
@@ -1212,6 +1218,13 @@ submodule openconfig-ospfv2-lsdb {
         "The mask of the network described by the Summary LSA
         represented as a CIDR mask.";
     }
+
+    leaf metric {
+      type oc-ospf-types:ospf-metric;
+      description
+        "The cost of utilising the link specified independent of TOS";
```
Comment (robshakir) [inline release/models/ospf/openconfig-ospfv2-lsdb.yang:1225]:
```suggestion
        "The cost of utilising the link specified independent of TOS.";
```

Comment (robshakir) [inline release/models/ospf/openconfig-ospfv2-lsdb.yang:1225]: Is this description consistent with where we have metric elsewhere in the OSPF model?

File: release/models/ospf/openconfig-ospfv2.yang [hunk 10]
```diff
@@ -34,7 +34,13 @@ module openconfig-ospfv2 {
     "An OpenConfig model for Open Shortest Path First (OSPF)
     version 2";
 
-  oc-ext:openconfig-version "0.4.2";
+  oc-ext:openconfig-version "0.5.0";
+
+  revision "2023-07-17" {
+    description
+      "add leaf metric to lsdb-summary-lsa-state";
```
Comment (robshakir) [inline release/models/ospf/openconfig-ospfv2.yang:41]: Final one to fix.

**Review by robshakir — CHANGES_REQUESTED** [review]

File: release/models/bgp/openconfig-bgp-errors.yang (hunk 1 above)
Comment (dplore) [reply release/models/bgp/openconfig-bgp-errors.yang]: reverted unintended changes to bgp

File: release/models/ospf/openconfig-ospfv2-lsdb.yang (hunk 9 above)
Comment (dplore) [reply release/models/ospf/openconfig-ospfv2-lsdb.yang]: The description is a cut and paste of the router LSA metric description.   I updated to acknowledge which container this metric is in to differentiate the description from the network LSA.

File: release/models/ospf/openconfig-ospfv2-lsdb.yang (hunk 8 above)
Comment (dplore) [reply release/models/ospf/openconfig-ospfv2-lsdb.yang]: Done

File: release/models/ospf/openconfig-ospfv2-global.yang (hunk 7 above)
Comment (dplore) [reply release/models/ospf/openconfig-ospfv2-global.yang]: Done

File: release/models/ospf/openconfig-ospfv2-area.yang (hunk 5 above)
Comment (dplore) [reply release/models/ospf/openconfig-ospfv2-area.yang]: Done

**Comment by dplore** [comment]
Ready for review

**Comment by dplore** [comment]
Thanks for the earlier review @robshakir, please take a look at the latest.

**Review by robshakir — APPROVED** [review]

**Comment by dplore** [comment]
Set last call for Aug 22
